profiles/views.py

```diff
@@ -1,3 +1,12 @@
-from django.shortcuts import render
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from .models import User
+from .serializers import UserSerializer
 
-# Create your views here.
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])  # Дозвіл для авторизованих користувачів
+def user_profile(request):
```

Comment:
I need class-based view here. Read everything about DRF ecosystem. Generally, views, serializers, viewsets, generic views, permissions, authenticators, responses, drf raise HTTP exceptions and filters .